src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -239,9 +239,14 @@ def __write_kv_and_features_to_config_store(cmd, key_values, features=None, name
         if features:
             key_values.extend(__convert_featureflag_list_to_keyvalue_list(features))
 
-        for kv in key_values:
-            kv.label = label
-            azconfig_client.set_keyvalue(kv, ModifyKeyValueOptions())
+        if not preserve_labels:
+            for kv in key_values:
+                kv.label = label
+                azconfig_client.set_keyvalue(kv, ModifyKeyValueOptions())
+        else:
+            for kv in key_values:
+                azconfig_client.set_keyvalue(kv, ModifyKeyValueOptions())
```

Comment:
```
        for kv in key_values:
            kv.label = label if not preserve_labels else kv.label
            azconfig_client.set_keyvalue(kv, ModifyKeyValueOptions())
```

Reply:
For better performance, I put this check outside the for loop. We don't need to check preserve_labels for each kv.

---
In reply to: [363904558](https://github.com/Azure/azure-cli/pull/11756#discussion_r363904558) [](ancestors = 363904558)